docs/v1/api.md

```diff
@@ -1,63 +1,93 @@
+an instance of a service class.  The `Instance` has a reference to the
+`ServiceClass` to provision an instance of.
 
-*TODO: what happens in the below cases?*
+The service catalog controller has a watch on the `Instance` resource and
+receives an `ADD` watch event. The controller then attempts to provision a new
+instance:
 
-1. *The `Broker` that caused the `ServiceClass` was deleted*
-1. *The `ServiceClass` itself was deleted*
+1.  The controller calls the provision endpoint on the broker 
+2.  The broker returns a response indicating that the broker provisioned the new
+    instance, the instance was already provisioned, or the provisioning
+    operation is in progress
+3.  The controller updates the status of the `Instance` to indicate when the
+    `Instance` is in a provisioning or provisioned condition
 
-## `Instance`
+*TODO: codify how asynchronous responses are handled in the controller*
 
-This resource is created by a service consumer to indicate their desire to
-provision a service. When the service catalog's controller event loop sees an
-`Instance` created, it calls the provision endpoint on the backing CF service
-broker and writes `provisioned` into a `status` field of the
-`Instance`.
+## `Binding` Resource
 
-*TODO: what happens when an `Instance` resource is deleted?*
+A `Binding` represents the binding between an application and an `Instance` of a
+`ServiceClass`.  
 
-## `Binding`
+Service consumers create `Binding` resources to indicate that an application
+should be bound to an `Instance`.  The `Binding` contains information about how
+the application wants to use the binding information such as:
 
-This resource is created by a service consumer to indicate that an application
-should be bound to an instance. When the service catalog's controller event
-loop sees a `Binding` created, it calls the bind endpoint on the backing CF
-service broker. When a successful response is returned, the event loop does
-the following:
+1.  The name of a Kubernetes core `Service` resource to provide a stable
+    endpoint for the application to use the `Instance` via
+2.  The name of a Kubernetes `Secret` resource to hold credentials necessary to
+    use the service
 
-1. Writes `bound` into a `status` field of the `Binding`
-1. Writes the contents of `credentials` map into a secret (naming of the secret
-   to be discussed later)
-1. Updates the `Binding`'s status section with the fully qualified path to the
-   aforementioned secret
+The service catalog controller has a watch on the `Binding` resource.  When the
+controller receives an `ADD` event for a new Binding, it attempts to bind
+against the service instance:
 
-*TODO: what happens when a `Binding` resource is deleted?*
+1.  The controller calls the binding endpoint on the broker
+2.  The broker returns a response containing credentials and coordinates for
+    the binding
+3.  The controller creates a Kubernetes `Service` with the given name with the
```

Comment:
I do agree additional design and clarity on the specifics of API conditions is needed.